File: fortran/src/main/java/de/irt/jfor/Cub.java
package de.irt.jfor;

/**
 * Created by IntelliJ IDEA.
 * User: stueken
 * Date: 20.10.17
 * Time: 11:54
 */
public class Cub extends Mat {

    public final int nz;

    public Cub(int nx, int ny, int nz) {
        super(nx, ny);
        if(nz<1)
            throw new IllegalArgumentException("invalid  array index");
        this.nz = nz;
    }

    protected int index(int ix, int iy, int iz) {
        return index(ix, iy) + nx*ny * bound(iz, nz, "nz");
    }
}

File: fortran/src/main/java/de/irt/kfor/Cub.kt
package de.irt.kfor

/**
 * version:     $Revision$
 * created by:  dst
 * created on:  18.01.2019 10:34
 * modified by: $Author$
 * modified on: $Date$
 */
interface Cub {
    val ni : Int
    val nj : Int
    val nk : Int

    val len : Int get() = ni*nj*nk

    fun index(i : Int, j : Int, k : Int) = i-1 + nj*((j) + nk*(k-1))
}